src/parser/jobs/ast/modules/ArcanaTracking/ArcanaSuggestions.tsx

```diff
@@ -56,172 +48,132 @@ interface SleeveIcon {
+	private play: Array<Action['id']> = []
+	private sleeveIcon: SleeveIcon = {}
 
 	//div privates for output
-	private history: DIVWindow[] = []
-	private lastDIVFalloffTime: number = 0
-	private divCast: number = 0
+	private history: DivinationWindow[] = []
+	private currentWindow: DivinationWindow | undefined = undefined
+	private divinationCast: number = 0
 	//end div privates
 
-	protected override init() {
-		PLAY.forEach(actionKey => {
-			this.PLAY.push(this.data.actions[actionKey].id)
-		})
+	override initialise() {
+		this.play = PLAY.map(actionKey => this.data.actions[actionKey].id)
 
-		this.SLEEVE_ICON = {
+		this.sleeveIcon = {
 			[SleeveType.NOTHING]: '',
 			[SleeveType.ONE_STACK]: this.data.statuses.SLEEVE_DRAW.icon,
 			[SleeveType.TWO_STACK]: 'https://xivapi.com/i/019000/019562.png',
 		}
 
 		//used for divination tracking
-		this.addEventHook('normalisedapplybuff', {to: 'player', abilityId: this.data.statuses.DIVINATION.id}, this.tryOpenWindow)
-		this.addEventHook('normalisedapplybuff', {by: 'player', abilityId: this.data.statuses.DIVINATION.id}, this.countDIVBuffs)
-		this.addEventHook('removebuff', {to: 'player', abilityId: this.data.statuses.DIVINATION.id}, this.tryCloseWindow)
-		this.addEventHook('cast', {by: 'player'}, this.onCast)
+		const divinationFilter = filter<Event>().type('statusApply').status(this.data.statuses.DIVINATION.id)
+		//this.addEventHook(divinationFilter.target(this.parser.actor.id), this.tryOpenWindow)
+		this.addEventHook(divinationFilter.source(this.parser.actor.id), this.countDivinationBuffs)
+		this.addEventHook(
+			filter<Event>()
+				.type('statusRemove')
+				.target(this.parser.actor.id)
+				.status(this.data.statuses.DIVINATION.id),
+			this.tryCloseWindow,
+		)
 
 		this.addEventHook('complete', this._onComplete)
 	}
 
 	//functions for divination buff counting
-	private countDIVBuffs(event: NormalisedApplyBuffEvent) {
+	private countDivinationBuffs(event: Events['statusApply']) {
 		// Get this from tryOpenWindow. If a window wasn't open, we'll open one.
-		const lastWindow: DIVWindow | undefined = this.tryOpenWindow(event)
+		// If it was already open (because another Astrologian went first), we'll keep using it
+		this.tryOpenWindow(event)
 
 		// Find out how many players we hit with the buff.
-		if (lastWindow) {
-			lastWindow.playersBuffed += event.confirmedEvents.filter(hit => this.parser.fightFriendlies.findIndex(f => f.id === hit.targetID) >= 0).length
+		if (this.currentWindow != null && !this.currentWindow.playersBuffed.includes(event.target) && this.actors.get(event.target).playerControlled) {
+			this.currentWindow.playersBuffed.push(event.target)
 		}
 	}
 
-	private tryOpenWindow(event: NormalisedApplyBuffEvent): DIVWindow | undefined {
-		const lastWindow: DIVWindow | undefined = _.last(this.history)
-
-		if (lastWindow && !lastWindow.end) {
-			return lastWindow
+	private tryOpenWindow(event: Events['statusApply']): DivinationWindow {
```

Comment:
No need for return here. TBF this functionality can be moved to countDivinationBuffs, since it's not called anywhere else.

Reply:
to keep in spirit of open/close window, I merged countDivinationBuffs into open window instead